Firestore/Swift/Source/Codable/DocumentSnapshot+ReadDecodable.swift

```diff
@@ -37,8 +38,10 @@ extension DocumentSnapshot {
                                  with serverTimestampBehavior: ServerTimestampBehavior = .none,
                                  decoder: Firestore.Decoder? = nil) throws -> T? {
     let d = decoder ?? Firestore.Decoder()
+    d.userInfo[documentRefUserInfoKey] = reference
```

Comment:
Do you think there is a different way we could handle this? This does not seem very discoverable.

Reply:
My hopes were that the end user does not have any need for discovering this, as it would be an implementation detail of the `Firestore` `Codable` additions.

Don't you think that it's realistic that the `Codable` overlays for swift provided by the SDK were sufficient, so that end users would never directly decode structures from Firestore 'manually'?

As a side note, I really think that the `@DocumentID` property wrapper (which the above aims to support) is an anti-pattern, as it ties your model to a very specific use case - but of course it must be supported here as it's already part of the API.